config/backend-config/backend-config.go

```diff
@@ -162,6 +163,13 @@ func pollConfigUpdate() {
 		if !ok {
 			statConfigBackendError.Increment()
 		}
+
+		//sorting the sourceJSON.
+		//json unmarshal does not guarantee order. For DeepEqual to work as expected, sorting is necessary
+		sort.Slice(sourceJSON.Sources[:], func(i, j int) bool {
+			return sourceJSON.Sources[i].ID < sourceJSON.Sources[j].ID
+		})
```

Comment:
Do we also need to sort destinations array inside each source?

Reply:
The response for /hostedWorkspaceConfig is same for consecutive requests and its in the following format.

{ "workspaceId1" : [Array of sources], 
   "workspaceId2" : [Array of sources] }

This response is unmarshalled into map[string][]SourceT and a consolidated Array of sources is created.

It is during unmarshal we are losing order.

So sorting the consolidated sources array created by us would suffice.